### ISSUE TITLE
Network: After unclustering a node, deleting the reference to that cluster node in the nodes dataset will throw error in v4.7

### ISSUE THREAD

**Issue by dragonzone** [almende/vis#1160]
I recently upgrade from v4.5.1 to v4.7 and notice that if I un-cluster a node and then later go delete/remove that reference in my nodes dataset, which I store all the nodes and clusters, I will get an error in vis.js (line 27566), saying "nodes[id] is undefined". How can I remove that cluster node from my dataset without errors?

**Comment by AlexDM0** [comment]
by opening the cluster? I'm not sure what you mean. If you open the cluster it should be removed from the nodes. Clusters should never be in a dataset to begin with. It sounds like you're touching the internals. Dont :)

**Comment by dragonzone** [comment]
I have my own list of nodes and edges datasets that are pass to the network to do the drawing. Whenever I create a cluster node, I add it to my list of nodes. Whenever it is un-cluster, I remove it from my list of nodes. It has been working fine before until this version.

**Comment by AlexDM0** [comment]
How do you add it? Without code I don't really know what's going on. Still sounds like you're using internals to me... I can't support use of internal datastructures.

Regards

**Comment by AlexDM0** [comment]
You should not be able to get a reference to that specific node through the public API. If you did get it through public methods, I'd be very curious how you've done so..

**Comment by dragonzone** [comment]
Yes, I'm using the public methods from your docs. Here's my pseudo-code:

```
if (network.isCluster(item)) {
	var nodesInCluster = network.getNodesInCluster(item);
	_.each(nodesInCluster, function(it) {
		var node = nodes.get(it);
		node[resolveNameId] = '';
		nodes.update(node);
		node = nodes.get(it);
	})
	network.openCluster(item);
	nodes.remove(item);
}
```
Before, it used to work fine if I do nodes.remove(item), but now that's where it throws the error.

**Comment by AlexDM0** [comment]
Hi,

I completely misunderstood. I thought you ment a pointer to the internal object instead of the id.

I tracked down the bug and fixed it on develop.

Regards

**Comment by dragonzone** [comment]
ok, thx, when is the next push & version?

**Comment by AlexDM0** [comment]
I don't know yet I'm afraid. It's busy here at the moment so we have not had the time to make a planning. It could be this Friday or the next.

For now you can remove the line that is throwing the error, it will not harm anything.

Regards

**Comment by AlexDM0** [comment]
fixed in 4.8
